src/builtins.ts

```diff
@@ -771,6 +772,44 @@ export function compileCall(
         }
       }
     }
+    case BuiltinSymbols.nameof: {
+      let value: string | null = null;
+      let resultType: Type | null = evaluateConstantType(compiler, typeArguments, operands, reportNode);
+      compiler.currentType = compiler.program.stringInstance.type;
```

Comment:
`ensureStaticString` below will set this in the non-error case, so this statement can be moved into the error case below.

Reply:
Okay. I see what you're saying. The type only needs to be added in the error case. I have this all set locally. Will submit soon.